### PR TITLE
feat: add adaptive refresh frequency option

### PR DESCRIPTION
**Pull request by hhh2210**
Implements the adaptive refresh decision record from #1739, as rewritten there. Posting the implementation so it can be judged against the record's acceptance criteria; I understand the merge gate is still your product sign-off on the record itself, so treat this as ready-when-you-are.

## What it does

A new "Adaptive" choice in the Refresh cadence picker, listed after the fixed intervals. Defaults are untouched and fixed/manual behave exactly as before.

The decision lives in `AdaptiveRefreshPolicy` (new file, 70 lines), a pure function from input to decision with the first-match-wins table from the record: Low Power Mode or thermal serious/critical gives 30 min; a menu opened within the last 5 minutes gives 2 min; within the hour, 5 min; under 4 hours, 15 min; otherwise (or with no menu open recorded) 30 min. Every input lands inside the 2 to 30 minute bounds, including negative/future ages from clock adjustments.

The only signal is one in-memory `lastMenuOpenAt` timestamp set in `menuWillOpen`. It is never persisted, resets on launch, and never triggers a refresh by itself. Each timer tick logs `reason=<code> delay=<n>s` under a new `adaptive-refresh` category and nothing else. The record's non-decisions are respected: no history, no per-account state, no prewarming, no scheduler changes, no ML, not the default.

One line outside the strictly additive scope: the fixed-interval timer branch was missing the post-sleep `Task.isCancelled` check, so changing frequency mid-sleep fired one spurious refresh (the cancel wakes the sleep early and `try?` swallows it). Reproduced in a test, fixed with the same guard the adaptive branch uses. Happy to split this into its own PR if you prefer.

## Update (9e497b5): interval-derived heuristics stay active in adaptive mode

Clawsweeper's review caught a real gap here, and Copilot flagged a log nit; both are fixed.

`RefreshFrequency.seconds` is nil for adaptive, and four call sites read that nil as "manual, no normal poll". The worst one: the reset-boundary scheduler from #1857 disabled itself entirely in adaptive mode. The others: the OpenAI web staleness interval fell to its 120s floor instead of scaling with the cadence, and the persistent-CLI-session idle windows collapsed to their 180s floor, which would churn CLI sessions between adaptive ticks.

The fix follows the review's recommended option. A new `UsageStore.normalRefreshIntervalForHeuristics()` resolves manual to nil (unchanged), fixed intervals to their seconds, and adaptive to the delay the policy would pick right now from live signals. Three call sites route through it. The fourth, `ProviderRegistry.specs`, runs before a `UsageStore` exists, so adaptive maps there to a named nominal constant (300s, the policy's warm tier) via `ProviderRegistry.nominalRefreshInterval(for:)`.

Regression coverage is the new `AdaptiveRefreshHeuristicsTests` suite (9 tests, 13 cases). Each of the four call sites has a test that goes red if it is reverted to `.seconds`; I verified that by applying each revert as a mutation and running the suite (one focused failure per call site, and mutating the helper's adaptive branch back to nil turns four tests red). The reset-boundary case runs through the real `refresh()` pipeline with a stubbed codex fetch, in both adaptive mode (schedules) and manual mode (still doesn't).

The log line is now exactly `reason=<code> delay=<n>s` with the category carrying the identification, as promised below (the first push had a redundant "adaptive refresh: " prefix; thanks Copilot).

## Runtime proof

Built this branch with `Scripts/package_app.sh debug` (0.38.1 build 95, bundle `com.steipete.codexbar.debug`, isolated defaults domain), set the cadence to Adaptive, and ran it:

```
$ log stream --level debug --style compact --predicate 'subsystem == "com.steipete.codexbar" AND category == "adaptive-refresh"'
Timestamp               Ty Process[PID:TID]
2026-07-04 04:29:01.313 Db CodexBar[33407:a7afc7] [com.steipete.codexbar:adaptive-refresh] reason=longIdle delay=1800s
2026-07-04 04:32:00.919 Db CodexBar[33407:a7afc7] [com.steipete.codexbar:adaptive-refresh] reason=recentInteraction delay=120s
2026-07-04 04:34:20.183 Db CodexBar[33407:a7afc7] [com.steipete.codexbar:adaptive-refresh] reason=recentInteraction delay=120s
```

Reading the three lines: launch has no menu-open history, so the first decision is `longIdle` (30 min). At 04:32 I opened the status menu and re-selected Adaptive in Settings; the cadence change restarts the timer, and the fresh decision sees the seconds-old menu open, so `recentInteraction` (2 min). The 04:34:20 line is the next tick firing on its own after the 120s sleep plus the refresh, recomputing from live signals. Nothing redacted; reason and delay are all these lines ever carry, by design.

The picker in Settings, running this build:

General settings pane with Refresh cadence set to Adaptive

The policy-table transitions themselves (warm/idle/constrained boundaries) are covered exhaustively in `AdaptiveRefreshPolicyTests` rather than re-proven against the wall clock.

## Tests

27 new cases across three files in the original push: the full policy table with boundary ages, priority ordering, and a bounds sweep over every input combination; timer cancellation tested in isolation from the settings-observer refresh path (restart at the same frequency, exact-delta assertions); and an integration test that drives a real `menuWillOpen` and checks the signal lands on the store. Mutation-checked: deleting either cancellation guard or the `noteMenuOpened()` call turns at least one test red. Plus the 13 heuristics cases described above.

## Validation

- `make check`: 0 violations in 1239 files
- `make test`: all 45 shards pass
- `swift build` and `swift build --build-tests`: clean
- All 21 locale catalogs gained a translated "Adaptive" label; `plutil -lint` clean, locale coverage suites pass
- `node Scripts/check-documentation-links.mjs`: OK, 142 local links

`docs/refresh-loop.md` gained an "Adaptive mode" section documenting the table, signal ownership, and the heuristics interval contract.

Refs #1739.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

https://claude.ai/code/session_01KmMSEXFQeQHSBZjUyK1hoU

### CODE REVIEW

**Comment by hhh2210** [comment]
@clawsweeper re-review

File: Sources/CodexBar/SettingsStore.swift [hunk 1]
```diff
@@ -23,6 +28,7 @@ enum RefreshFrequency: String, CaseIterable, Identifiable {
         case .fiveMinutes: 300
         case .fifteenMinutes: 900
         case .thirtyMinutes: 1800
+        case .adaptive: nil
         }
```
[the remark this answers is not in the document]
Comment (steipete) [reply Sources/CodexBar/SettingsStore.swift:32]: Fixed on head `30f2de7f`.

`RefreshFrequency.seconds` remains the fixed-interval value, while `UsageStore.normalRefreshIntervalForHeuristics()` now computes the live Adaptive delay in `Sources/CodexBar/UsageStore+AdaptiveRefresh.swift:67`. Reset-boundary scheduling, OpenAI web staleness, and CLI idle-window consumers all use that helper (`UsageStore.swift:708`, `UsageStore+OpenAIWeb.swift:69`, and `UsageStore+TokenAccounts.swift:718`). `AdaptiveRefreshHeuristicsTests` covers every migrated consumer.

File: Sources/CodexBar/UsageStore+AdaptiveRefresh.swift [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+            now: now,
+            lastMenuOpenAt: lastMenuOpenAt,
+            lowPowerModeEnabled: lowPowerModeEnabled,
+            thermalState: thermalState))
+    }
+
+    func logAdaptiveRefreshDecision(_ decision: AdaptiveRefreshPolicy.Decision) {
+        // Reason and delay only; never provider/account/email/path/credential/response data.
+        self.adaptiveRefreshLogger.debug(
+            "adaptive refresh: reason=\(decision.reason.rawValue) delay=\(decision.delay.components.seconds)s")
```
[the remark this answers is not in the document]
Comment (steipete) [reply Sources/CodexBar/UsageStore+AdaptiveRefresh.swift]: Fixed on head `30f2de7f` at `Sources/CodexBar/UsageStore+AdaptiveRefresh.swift:37`: the category now logs only `reason=<code> delay=<n>s`. Focused tests and signed live logs confirm the exact format.

**Comment by steipete** [comment]
Landed after hardening the adaptive timer lifecycle and validating the exact contributor head `30f2de7fd1af2cbdcb7d534a5c32f835e1beedf8`.

Proof:

- 28 focused adaptive-refresh tests across 3 suites passed.
- `make test` passed all 46/46 shards.
- `make check` passed.
- Autoreview returned no accepted/actionable findings.
- Signed local app proof: selecting Adaptive produced the expected long-idle decision, `reason=longIdle delay=1800s`; the prior five-minute setting was restored afterward.
- Exact-head CI passed all 10/10 checks.

Thanks again @hhh2210!
